datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1520,9 +1639,79 @@ impl SMJStream {
         } else {
             self.output_size -= record_batch.num_rows();
         }
-        self.output_record_batches.clear();
+        if self.filter.is_none() || self.join_type != JoinType::Left {
+            self.output_record_batches.batches.clear();
```

Comment:
for filtered joins we dont need to clear batches as we need them later, I'll do it rhough the flag later